google-datacatalog-qlik-connector/src/google/datacatalog_connectors/qlik/scrape/engine_api_helper.py

```diff
@@ -0,0 +1,161 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+import json
+
+from urllib.parse import urlparse
+import websockets
+
+from google.datacatalog_connectors.qlik.scrape import constants
+
+
+class EngineAPIHelper:
+    """Wraps requests to the Qlik Qlik Engine JSON API.
+
+    The Qlik Engine JSON API is a WebSocket protocol that uses JSON to pass
+    information between the Qlik associative engine and the clients. This API
+    consists of a set of objects representing apps, lists, and so on. These
+    objects are organized in a hierarchical structure.
+
+    Websockets use an asynchronous communication channel, but the public
+    methods from this class are intended to be called synchronously. They
+    take care of handling the async API calls for their clients.
+
+    """
+
+    def __init__(self, server_address):
+        self.__server_address = server_address
+        # The server address starts with an http/https scheme. The below
+        # statement replaces the original scheme with 'wss', which is used for
+        # secure websockets communication.
+        self.__base_api_endpoint = \
+            f'wss://{self.__extract_server_netloc(server_address)}'
+        self.__common_headers = {
+            constants.XRFKEY_HEADER_NAME: constants.XRFKEY,
+        }
+
+    @classmethod
+    def __extract_server_netloc(cls, server_address):
+        parsed_uri = urlparse(server_address)
+        netloc = parsed_uri.netloc
+        netloc_colon_index = netloc.find(':')
+
+        if netloc_colon_index > 0:
+            # Strip the port number.
+            return netloc[:netloc_colon_index]
+
+        return netloc
+
+    def get_sheets(self, app_id, auth_cookie):
+        """Get the list of Sheets that belong to a given App.
+
+        Returns:
+            A list of sheets.
+        """
+        return self.__run_until_complete(self.__get_sheets(
+            app_id, auth_cookie))
+
+    async def __get_sheets(self, app_id, auth_cookie):
+        websocket = await self.__connect_websocket(app_id, auth_cookie)
+        open_doc_return = await self.__open_doc(app_id, websocket)
+
+        await websocket.send(
+            json.dumps({
+                'handle': open_doc_return.get('qHandle'),
+                'method': 'GetObjects',
+                'params': {
+                    'qOptions': {
+                        'qTypes': ['sheet'],
+                    },
+                },
+            }))
+
+        async for message in websocket:
```

Comment:
Is this `auth_cookie` reused in other scrape methods or connector executions?

Let me try to put together a use case to go in more detail where I want to go with that question... 

Let's say the `qlik` connection is scheduled to execute every 30 minutes... For some reason the 1st execution takes longer and it overlaps with the second execution...

If this `auth_cookie` is generated with the same value (I don't know if it's possible) but if it is, then this async iterator on the websocket could receive a result from others executions, does this make sense?

Is there a way to locate the result using the request identifier?